packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -452,6 +456,32 @@ export abstract class FluidDataStoreContext
 
 		const channel = await factory.instantiateDataStore(this, existing);
 		assert(channel !== undefined, 0x140 /* "undefined channel on datastore context" */);
+
+		//* TODO: Properly plumb this bit from ContainerRuntime
+		const gcEnabled = (this._containerRuntime as any).garbageCollector?.shouldRunGC ?? true; //* Some tests mock CR and this isn't there.
```

Comment:
I wonder what percentage of attach ops are realized immediately after they arrive. I assume in the summarizer client, that doesn't happen very frequently.

I really don't like that this has to be in the realize function. I would prefer a synchronous process over an asynchronous "processing" of the attach op.

Reply:
Another side note, does the local datastore context need this functionality? Technically it's already attached and we could just run getGCData on it. Just a thought.

Reply:
> does the local datastore context need this functionality

Had to think about it for a minute.

Summarizer is the main customer here, and it "shouldn't" have local changes.  But it can today, so in that case we need to do something.  Need to look into how the attach flow works for local things when the attach op is ack'd

(Interactive clients benefit as well, in that they clear unreferenced state when notified of a new reference, which reduces false positives of warning logs.)

Reply:
@agarwal-navin - thoughts on local channels? I need to address this in either case. I'll look into it soon but if you already can see what needs to be done, feel free to chime in :)

Reply:
Resolving this thread.

- Switched to the snapshot-based approach rather than getting GC data during realize
- With that, we are pulling the GC Data from the snapshot synchronously when the attach op is processed, so we can easily do that before branching off for local v. remote messages.